### PR TITLE
minor improments on withConnection method usage

### PR DESCRIPTION
**Pull request by octonato**
Draft because on top of #137.

Ready for review otherwise.

### CODE REVIEW

File: core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala [hunk 1]
```diff
@@ -138,74 +138,21 @@ class R2dbcExecutor(val connectionFactory: ConnectionFactory, log: Logger)(impli
...
-        rowsUpdated.failed.foreach { exc =>
-          log.debug("{} - Update failed: {}", logPrefix, exc)
-          // auto-commit so nothing to rollback
-          connection.close().asFutureDone()
-        }
-
-        rowsUpdated.flatMap { r =>
-          connection.close().asFutureDone().map { _ =>
-            if (log.isDebugEnabled())
-              log.debug("{} - Updated [{}] in [{}] µs", logPrefix, r, (System.nanoTime() - startTime) / 1000)
```
Comment (octonato) [inline core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala:202]: There is not much different in terms of logging, so reusing new `withAutoCommitConnection`

File: core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala [hunk 2]
```diff
@@ -138,74 +138,21 @@ class R2dbcExecutor(val connectionFactory: ConnectionFactory, log: Logger)(impli
...
-            r
-          }
-        }
-      }
+  def updateOne(logPrefix: String)(statementFactory: Connection => Statement): Future[Int] =
+    withAutoCommitConnection(logPrefix) { connection =>
```
Comment (octonato) [inline core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala:153]: we could eventually add `logPrefix + "(UpdateOne)"` just to add more context.

File: core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala [hunk 3]
```diff
@@ -138,74 +138,21 @@ class R2dbcExecutor(val connectionFactory: ConnectionFactory, log: Logger)(impli
...
-
-        done.flatMap { _ =>
-          connection.close().asFutureDone().map { _ =>
-            if (log.isDebugEnabled())
-              log.debug("{} - DDL in [{}] µs", logPrefix, (System.nanoTime() - startTime) / 1000)
-            Done
-          }
-        }
+  def executeDdl(logPrefix: String)(statement: Connection => Statement): Future[Done] =
+    withAutoCommitConnection(logPrefix + " (DDL)") { connection =>
```
Comment (octonato) [inline core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala:142]: Enriched prefix for extra context. Maybe something we should do each time.

**Comment by octonato** [comment]
Actually, I'm not so sure anymore if we should append even more information on the logPrefix like I did here with the DDL. 
At the end, the original log prefix should contain enough info. If not, then we are not using is as it should. 

Looking for other opinions.

File: core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala (hunk 3 above)
Comment (octonato) [reply core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala]: I removed this extra prefix info. Not convinced is a good idea.

**Comment by patriknw** [comment]
> Actually, I'm not so sure anymore if we should append even more information on the logPrefix like I did here with the DDL.
At the end, the original log prefix should contain enough info. If not, then we are not using is as it should.

I think I agree that it's better to let the caller define the full information.

**Review by patriknw — APPROVED** [review]
LGTM, great cleanup

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by octonato** [comment]
no clear what happened here. Something closed this PR and now Lightbend CLA validator is not even recognizing me anymore. 🤦 

Anyway, no time for this now. Tomorrow I check it again.

**Pull request closed without merging**